src/qml/qmlwaveformrenderer.h

```diff
@@ -71,9 +79,10 @@ class QmlWaveformRendererPreroll
     ::WaveformRendererAbstract::PositionSource m_position{::WaveformRendererAbstract::Play};
 };
 
-class QmlWaveformRendererRGB
+class QmlWaveformRendererSignal
         : public QmlWaveformRendererFactory {
     Q_OBJECT
+    Q_PROPERTY(bool ignoreStem MEMBER m_ignoreStem NOTIFY ignoreStemChanged)
     Q_PROPERTY(QColor axesColor MEMBER m_axesColor NOTIFY axesColorChanged REQUIRED)
     Q_PROPERTY(QColor lowColor MEMBER m_lowColor NOTIFY lowColorChanged REQUIRED)
     Q_PROPERTY(QColor midColor MEMBER m_midColor NOTIFY midColorChanged REQUIRED)
```

Comment:
Can you clarify why you moved all of the color stuff out the RGB renderer into the WaveformRendererSignal?

Reply:
Sure, since colours are managed centrally in `waveform/renderers/allshader/waveformrenderersignalbase.cpp`, and then reused in all renderers, I thought it would make sense to replicate this separation of concern in QML, so changes to either could be synchronised, rather than having redundant reference between each renderer in QML and that `WaveformRendererSignalBase`

Reply:
right, it makes sense why QML mirrors C++, my question is that I don't understood why it was getting centralized in the first place since there are also renderers don't have this concept of high mid and low colors, right?

Reply:
I'm not sure I follow. Let me rephrase my question:

The WaveformRendererSignalBase should have no notion of color, at least when there are childclasses such as WaveformRendererFiltered that don't. So why is this moved into the SignalBase?

Reply:
I'm confused - are you asking about `WaveformRendererSignalBase` or `mixxx:qml:QmlWaveformRendererSignal`?

Reply:
> The QmlWaveformRenderer* class hierarchy mirrors the C++ WaveformRenderer*, right?

That's right, but the color are not handle inside `WaveformRenderer*`, but in `WaveformRendererSignalBase` instead.

Are you asking me to copy-paste these property on each renderers where relevant?

Reply:
> That's right, but the color are not handle inside WaveformRenderer*, but in WaveformRendererSignalBase instead.

Oh really? I need to brush up on my knowledge on that class again. I'll try to find some time to refamiliarize myself with it again.

> Are you asking me to copy-paste these property on each renderers where relevant?

No, I just care about separation of concerns, but if the class already violates that, then my concerns are void (or at least out-of-scope).